pkg/costattribution/active_tracker.go

```diff
@@ -169,19 +237,29 @@ func (at *ActiveSeriesTracker) Collect(out chan<- prometheus.Metric) {
 	at.observedMtx.RLock()
 	if !at.overflowSince.IsZero() {
 		var activeSeries int64
+		var nativeHistogram int64
+		var nhBucketNum int64
 		for _, as := range at.observed {
-			activeSeries += as.Load()
+			activeSeries += as.asCount.Load()
+			nativeHistogram += as.nhCount.Load()
+			nhBucketNum += as.nhBucketNum.Load()
 		}
 		at.observedMtx.RUnlock()
-		out <- prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(activeSeries+at.overflowCounter.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+		out <- prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(activeSeries+at.overflowCounter.asCount.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+		out <- prometheus.MustNewConstMetric(at.activeNativeHistogramSeriesPerUserAttribution, prometheus.GaugeValue, float64(nativeHistogram+at.overflowCounter.nhCount.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+		out <- prometheus.MustNewConstMetric(at.activeNativeHistogramBucketsPerUserAttribution, prometheus.GaugeValue, float64(nhBucketNum+at.overflowCounter.nhBucketNum.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
 		return
 	}
 	// We don't know the performance of out receiver, so we don't want to hold the lock for too long
 	var prometheusMetrics []prometheus.Metric
 	for key, as := range at.observed {
 		keys := strings.Split(key, string(sep))
 		keys = append(keys, at.userID)
-		prometheusMetrics = append(prometheusMetrics, prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(as.Load()), keys...))
+		prometheusMetrics = append(prometheusMetrics,
+			prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(as.asCount.Load()), keys...),
+			prometheus.MustNewConstMetric(at.activeNativeHistogramSeriesPerUserAttribution, prometheus.GaugeValue, float64(as.nhCount.Load()), keys...),
```

Comment:
Always adding the native histogram series and buckets triples the number of series even for tenant that don't use native histograms. I think we should only append them is series count > 0. Of course this would need a new test case where we go from no native series to a couple then back 0 and check that the two additional series come and go.

I assume it that same for the basic series attribution as well, if the attributed series go away, we remove the corresponding `cortex_ingester_attributed_active_series` series !?

Reply:
At first, I wasn't sure how recording rules would behave when evaluating active_series{attribution='a'} - histogram{attribution='a'} + 0.5 * histogram_buckets{attribution='a'} if histogram and histogram_buckets were missing. If the result treated missing metrics as 0, then it would work fine, but otherwise, it might lead to an empty result. Taking a second thought, we could work around this by using active_series{attribution='a'} - (histogram{attribution='a'} or 0) + 0.5 * (histogram_buckets{attribution='a'} or 0) in the worst case. let me remove it for now.

Reply:
> if the attributed series go away, we remove the corresponding cortex_ingester_attributed_active_series series !?
correct, that attribution would be purged from map (cache), so yes metrics would be gone.

Reply:
that should work, i removed those 2 native histogram metrics when it is zero accordingly.